cmd/juju/application/deployer/charm.go

```diff
@@ -259,11 +258,8 @@ func (d *deployCharm) deploy(
 		return nil
 	}
 
-	if pErr, ok := errors.Cause(err).(*params.Error); ok {
-		switch pErr.Code {
-		case params.CodeAlreadyExists:
-			return errors.Wrapf(err, errors.Errorf("\ndeploy application using an alias name, or use remove-application to remove the existing one and try again"), err.Error())
-		}
+	if errors.IsAlreadyExists(err) {
+		return errors.Wrapf(err, errors.Errorf("\ndeploy application using an alias name, or use remove-application to remove the existing one and try again"), err.Error())
```

Comment:
Minor nit; I think you can get rid of the `\n` so it reads as one line after the `:` when the error is printed out.

Reply:
The error message is huge otherwise...